src/jdk.hotspot.agent/macosx/native/libsaproc/MacosxDebuggerLocal.m

```diff
@@ -50,8 +50,14 @@
 #define amd64 1
 #endif
 
-#if amd64
+#if defined(__arm64__) && !defined(aarch64)
```

Comment:
Have you tried to compile the native part if sa_agent ( libsaproc) on mac-intel?

I supposed you may get some issue with undefined preprocessor values

Seeing some issues here in SA part of this patch, I supposed we take azul's version ( when we can share it, due to xcode12beta restriction), it was tested and reviewed internally.